src/live_restore/live_restore_fs.c

```diff
@@ -633,11 +644,32 @@ __live_restore_fh_write(
     lr_fh = (WTI_LIVE_RESTORE_FILE_HANDLE *)fh;
     session = (WT_SESSION_IMPL *)wt_session;
 
+    /*
+     * Fast path writes if the destination is complete. This pointer is only ever cleared in a
+     * multithreaded context, if we read a valid pointer we will take the slow path with the same
```

Comment:
What do we mean by "only ever cleared in a multithreaded context"? Is `lr_fh->source` a shared variable that needs atomic accesses/thread safe clearing?

Reply:
Discussed this offline. The fast path is done outside of the file lock, but we take the lock and then re-check WTI_DEST_COMPLETE inside the lock meaning this is thread safe. 
The fast path isn't thread safe, but since the life time of the source file handle is only ever non-NULL => NULL seeing an out of date pointer means we'll enter the locked, safe path.